### PR TITLE
Gh 21567

### PR DESCRIPTION
**Pull request by srisri332**
#### Summary

Add the request context and logger to all public methods in server/channels/app/authorization.go

#### Ticket Link

Github Issue: https://github.com/mattermost/mattermost/issues/21567

#### Screenshots
 NA

#### Release Note

```
Refactored the methods listed down to accept rctx request.CTX as the first argument. 
```

PS: Go to the following links to find changes made for each method and their corresponding files where the methods are called. 
 - [SessionHasPermissionToUser](https://github.com/srisri332/mattermost/commit/18879c974bdd76560ca65d9128d52e46158f1a7a)
- [HasPermissionToUser](https://github.com/srisri332/mattermost/commit/b2469e24343db84916af53e7387c3b840ae32ff7)
- [HasPermissionTo](https://github.com/srisri332/mattermost/commit/b3b84bc73c3f2565fe58c41cc019b0e6874fd106)
- [SessionHasPermissionToChannelByPost](https://github.com/srisri332/mattermost/commit/95cc505c6a628cdb0dabbca9ef6714c4155b1037)
- [SessionHasPermissionToGroup](https://github.com/srisri332/mattermost/commit/4b833169f2fd57885d938171146cf95996c992ec)
- [SessionHasPermissionToAny](https://github.com/srisri332/mattermost/commit/e2a16adcdaf9d2bd3f9641ead4b9d48be1ea2923)

### CODE REVIEW

File: server/channels/app/authorization.go [hunk 1]
```diff
@@ -21,7 +21,7 @@ func (a *App) SessionHasPermissionTo(session model.Session, permission *model.Pe
 	return a.RolesGrantPermission(session.GetUserRoles(), permission.Id)
 }
 
-func (a *App) SessionHasPermissionToAny(session model.Session, permissions []*model.Permission) bool {
+func (a *App) SessionHasPermissionToAny(c request.CTX, session model.Session, permissions []*model.Permission) bool {
```
Comment (hanzei) [inline server/channels/app/authorization.go:24]: `request.CTX` already contains a Session. I wonder if we could simply drop the `session` argument.

@isacikgoz What do you think?

**Review by hanzei — CHANGES_REQUESTED** [review]
Thanks for the contribution!

File: server/channels/app/authorization.go (hunk 1 above)
Comment (srisri332) [reply server/channels/app/authorization.go:24]: @hanzei , Should I change this or wait for confirmation?

**Comment by hanzei** [comment]
@srisri332 Upon re-reviewing the ticket, I noticed that none of the methods in `server/channels/app/authorization.go` actually use `request.Context`. Hence, it's not necessary at the moment to add it do the methods.

I'm going to close this PR and the related ticket. Please excuse the confusion. Still, thanks for the contribution @srisri332

**Pull request closed without merging**